mkosi/kmod.py

```diff
@@ -79,19 +148,38 @@ def filter_firmware(
         firmwared = Path("usr/lib/firmware")
         with chdir(root):
             all_firmware = set(firmwared.rglob("*"))
+
+        patterns = [p[3:] for p in include if p.startswith("re:")]
         regex = re.compile("|".join(include))
+
         for f in all_firmware:
             rel = os.fspath(Path(*f.parts[3:]))
-            if regex.search(rel):
+            if (patterns and regex.search(rel)) or globs_match_firmware(rel, globs):
                 firmware.add(f)
 
+    logging.debug(f"Including {len(firmware)} firmare files")
+
     return firmware
 
 
 def normalize_module_name(name: str) -> str:
+    # Replace '_' by '-'
     return name.replace("_", "-")
 
 
+def normalize_module_glob(name: str) -> str:
+    # We want to replace '_' by '-', except when used in […]
+    ans = ""
+    while name:
+        i = (name + "[").index("[")
+        ans += name[:i].replace("_", "-")
+        name = name[i:]
+        i = (name + "]").index("]")
+        ans += name[:i + 1]  # fmt: skip
+        name = name[i + 1:]  # fmt: skip
```

Comment:
Please, let's not do this
```suggestion
        ans += name[: i + 1]
        name = name[i + 1 :]
```
Do I find the above a bit too airy, sure, but it's what [PEP8](https://peps.python.org/pep-0008/#pet-peeves) would recommend and we shouldn't proliferate `# fmt: skip`s beyond where they are really needed for better readability and I don't think this is it.

Reply:
This is another case where the automatic formatter does the wrong thing and we accept that. We don't actually follow PEP8 all that closely, almost nobody does. But OK, in this case the automatic version is just slightly worse, not egregious and completely unreadable like in many other cases.